azurelinuxagent/common/event.py

```diff
@@ -339,12 +342,27 @@ def add_metric(self, category, counter, instance, value, log_event=False):
         event.parameters.append(TelemetryEventParam('Instance', instance))
         event.parameters.append(TelemetryEventParam('Value', value))
 
+        self._add_default_parameters_to_event(event, container_id=os.environ.get(
+            CONTAINER_ID_ENV_VARIABLE, "UNINITIALIZED"), event_time_opcode_name=event_time_opcode_name,
+                                              event_pid=event_pid, event_tid=event_tid, task_name=task_name,
+                                              keyword_name=keyword_name)
         data = get_properties(event)
         try:
             self.save_event(json.dumps(data))
         except EventError as e:
             logger.error("{0}", e)
 
+    @staticmethod
+    def _add_default_parameters_to_event(event, container_id="UNINITIALIZED", event_time_opcode_name=None, event_pid=0,
+                                         event_tid=0, task_name='', keyword_name=''):
+        event.parameters.append(TelemetryEventParam('ContainerId', container_id))
+        event.parameters.append(TelemetryEventParam('OpcodeName', event_time_opcode_name if event_time_opcode_name
+                                                    else datetime.utcnow().__str__()))
+        event.parameters.append(TelemetryEventParam('EventTid', event_tid))
+        event.parameters.append(TelemetryEventParam('EventPid', event_pid))
+        event.parameters.append(TelemetryEventParam("TaskName", task_name))
```

Comment:
for task/keyword, which we currently do not set, should the default be something else than empty? is it important to distinguish whether they are populated vs whether they are not provided?

Reply:
For Task, I was thinking of "ExtHandler" (the prefix we use for Logger as well) as the default TaskName.

> is it important to distinguish whether they are populated vs whether they are not provided?

The necessity of providing at least empty values is to make sure that MA gets the schema correctly on the host side.